### PR TITLE
Adding Schema for Video Block in Sidebar 

### LINKED ISSUE #2769: Enable blocks enhancers in stock blocks

### PR DESCRIPTION
**Pull request by iRohitSingh**
Fixes #2769

### CODE REVIEW

**Comment by nileshgulia1** [comment]
@iRohitSingh Pls rebase with master to make it green!

**Review by sneridagh — CHANGES_REQUESTED** [review]
Sync with master,  and 

@iRohitSingh we need the block to be extendable too, using:

```
import { withBlockExtensions } from '@plone/volto/helpers';

withBlockExtensions(...)
```

File: src/config/Blocks.jsx [hunk 1]
```diff
@@ -300,6 +301,9 @@ const blocksConfig = {
     view: ViewVideoBlock,
     edit: EditVideoBlock,
     schema: BlockSettingsSchema,
+    blockSchema: VideoBlockSchema,
+    extensions: {},
```
Comment (sneridagh) [inline src/config/Blocks.jsx:305]: This is not needed.

Comment (sneridagh) [reply src/config/Blocks.jsx:305]: nor variations.

**Review by sneridagh — CHANGES_REQUESTED** [review]
Super small changes, remove unnecessary options.

The rest LGTM. 

@ichim-david any other last moment addition?

File: src/components/manage/Blocks/Video/schema.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    {
+      id: 'default',
+      title: 'Default',
+      fields: ['url', 'preview_image', 'align'],
+    },
+  ],
+
+  properties: {
+    url: {
+      title: props.intl.formatMessage(messages.VideoURL),
```
Comment (sneridagh) [inline src/components/manage/Blocks/Video/schema.js:34]: Just realised that the original has an action item with the X to remove the contents of the field:

image

In fact, it happens also in the new image block refactor /cc @nileshgulia1 @ThomasKindermann 

I'd like to maintain it somehow... maybe add it to the URLWidget?

**Comment by ichim-david** [comment]
> io

preview_image_error

I'm testing the branch again and when I've tried to use a preview image from the images that are already  uploaded using the tree selector I get the following error so this isn't stable yet.

**Comment by ichim-david** [comment]
> > io
> 
> preview_image_error @sneridagh I'm testing the branch again and when I've tried to use a preview image from the images that are already uploaded using the tree selector I get the following error so this isn't stable yet.

@sneridagh 
When using the object_browser on schema it seems that the value is an array of items
preview_image_array

When using it as part of the textWidget the data.preview is directly basically the id of the image 
https://github.com/plone/volto/blob/master/src/components/manage/Blocks/Video/VideoSidebar.jsx#L103

I don't have time to fix this now but with some checking someone should be able to figure out how to fix it pretty quickly

**Comment by ichim-david** [comment]
@sneridagh once that is fixed and your issues with the x (I think that comes from the different widgets, the text vs object_browser) it's fine for me to merge

**Comment by nileshgulia1** [comment]
> When using the object_browser on schema it seems that the value is an array of items

@ichim-david It seems I'm facing the same problems with using objectbrowserWidget as per https://github.com/plone/volto/pull/2700#issuecomment-1120259223
I solved it by replacing it with UrlWidget which keeps "url" field as string and also has support for "X" as clear button.

**Comment by ichim-david** [comment]
> @ichim-david It seems I'm facing the same problems with using objectbrowserWidget as per https://github.com/plone/volto/pull/2700#issuecomment-1120259223
> I solved it by replacing it with UrlWidget which keeps "url" field as string and also has support for "X" as clear button.
> 
@nileshgulia1  the URL widget seems to solve the issue indeed.
@sneridagh is this an ok change to use another widget?

There is still an issue that I found though when you add a youtube video and you add a preview image the video id
is missed and you end up with a playback error.
These issues scream of tests needed to me :) 
playback-error

File: src/config/Blocks.jsx (hunk 1 above)
Comment (nileshgulia1) [reply src/config/Blocks.jsx:305]: @danielamormocea could you please take care?

**Review by sneridagh — APPROVED** [review]

**Comment by sneridagh** [comment]
@ichim-david all good now with this one?

**Comment by ichim-david** [comment]
@sneridagh I saw that @danielamormocea  fixed the preview issue, if that fix is ok to have as it is (for youtube only, I didn't check other services to see if it happens for vimeo playback) then I am fine with merging it as all of the issues have been addressed

**Review by ichim-david — APPROVED** [review]

File: src/components/manage/Blocks/Video/VideoSidebar.jsx [hunk 3]
```diff
@@ -1,218 +1,51 @@
...
-                    onChangeBlock(block, {
-                      ...data,
-                      openLinkInNewTab: value,
-                    });
-                  }}
-                />
-              </Accordion.Content>
-            </Accordion>
-          )}
-        </>
```
Comment (giuliaghisini) [inline src/components/manage/Blocks/Video/VideoSidebar.jsx:204]: All this functionalities disappears! They are not added to the schema! this is breaking! please include it and do not remove functionalities

**Review by giuliaghisini — CHANGES_REQUESTED** [review]

File: src/components/manage/Blocks/Video/VideoSidebar.jsx (hunk 3 above)
Comment (sneridagh) [reply src/components/manage/Blocks/Video/VideoSidebar.jsx:204]: ups, they somehow slipped away! Sure, we must include them.

Comment (nileshgulia1) [reply src/components/manage/Blocks/Video/VideoSidebar.jsx:204]: @danielamormocea

**Comment by sneridagh** [comment]
Could someone take care of this?

File: src/components/manage/Blocks/Video/VideoSidebar.jsx (hunk 3 above)
Comment (nileshgulia1) [reply src/components/manage/Blocks/Video/VideoSidebar.jsx:204]: @giuliaghisini @sneridagh We don't use this code in the current https://6.demo.plone.org/xv. I think that code is obsolete and meant to be cleaned up.

Comment (giuliaghisini) [reply src/components/manage/Blocks/Video/VideoSidebar.jsx:204]: > 

yess.. the code there is! 
You have to link an mp4 video. 

Here an example:
https://6.demo.plone.org/test-video

Comment (nileshgulia1) [reply src/components/manage/Blocks/Video/VideoSidebar.jsx:204]: okay for `.mp4`. Thank you!. 
-cc @danielamormocea

**Comment by iFlameing** [comment]
@sneridagh I checked everything works perfectly fine

**Pull request merged**